lib/Utils/PlatformWin.cpp

```diff
@@ -0,0 +1,687 @@
+
+    const char *keyEnd = placeHolder - 1;
+    const char *nextKey = placeHolder;
+    // Find end of previous key.
+    while ((keyEnd > subKey) && (*keyEnd != '\\'))
+      keyEnd--;
+    // Find end of key containing $VERSION.
+    while (*nextKey && (*nextKey != '\\'))
+      nextKey++;
+    size_t partialKeyLength = keyEnd - subKey;
+    char partialKey[256];
+    if (partialKeyLength > sizeof(partialKey))
+      partialKeyLength = sizeof(partialKey);
+    ::strncpy(partialKey, subKey, partialKeyLength);
+    partialKey[partialKeyLength] = '\0';
+    HKEY hTopKey = NULL;
+    lResult = ::RegOpenKeyExA(hRootKey, partialKey, 0,
+                              KEY_READ | KEY_WOW64_32KEY, &hTopKey);
+    if (lResult == ERROR_SUCCESS) {
+      char keyName[256];
+      // int bestIndex = -1;
+      double bestValue = 0.0;
+      DWORD size = sizeof(keyName) - 1;
+      for (DWORD index = 0; ::RegEnumKeyExA(hTopKey, index, keyName, &size, NULL,
+                                            NULL, NULL, NULL) == ERROR_SUCCESS;
+           index++) {
+        const char *sp = keyName;
+        while (*sp && !isdigit(*sp))
+          sp++;
+        if (!*sp)
+          continue;
+        const char *ep = sp + 1;
+        while (*ep && (isdigit(*ep) || (*ep == '.')))
+          ep++;
+        char numBuf[32];
+        ::strncpy(numBuf, sp, sizeof(numBuf) - 1);
+        numBuf[sizeof(numBuf) - 1] = '\0';
+        double dvalue = ::strtod(numBuf, NULL);
+        if (dvalue > bestValue) {
+          // Test that InstallDir is indeed there before keeping this index.
+          // Open the chosen key path remainder.
+          ::strcpy(bestName, keyName);
+          // Append rest of key.
+          ::strncat(bestName, nextKey, sizeof(bestName) - 1);
+          bestName[sizeof(bestName) - 1] = '\0';
+          lResult = ::RegOpenKeyExA(hTopKey, bestName, 0,
+                                    KEY_READ | KEY_WOW64_32KEY, &hKey);
+          if (lResult == ERROR_SUCCESS) {
+            if (readFullStringValue(hKey, valueName, outValue)) {
+              // bestIndex = (int)index;
+              bestValue = dvalue;
+              returnValue = true;
+            }
+            ::RegCloseKey(hKey);
+          }
+        }
+        size = sizeof(keyName) - 1;
+      }
+      ::RegCloseKey(hTopKey);
+    } else
+      ReportError(lResult, "RegOpenKeyEx");
+  } else {
+    // If subKey is empty, then valueName is subkey, and we retreive that
+    if (subKey[0]==0) {
+      subKey = valueName;
+      valueName = nullptr;
+    }
+    lResult = ::RegOpenKeyExA(hRootKey, subKey, 0, KEY_READ | KEY_WOW64_32KEY,
+                              &hKey);
+    if (lResult == ERROR_SUCCESS) {
+      returnValue = readFullStringValue(hKey, valueName, outValue);
+      ::RegCloseKey(hKey);
+    } else
+      ReportError(lResult, "RegOpenKeyEx");
+  }
+  return returnValue;
+}
+
+int GetVisualStudioVersionCompiledWith() {
+#if (_MSC_VER >= 1900)
+  return 14;
```

Comment:
In either case it will still be broken for the forthcoming Visual Studio "15". And I don't want to guess what will happen. (https://en.wikipedia.org/wiki/Microsoft_Visual_Studio)

Originally there was no discrimination and it was _**(_MSC_VER / 100) - 6**_ in some places.